README.md

```diff
@@ -1 +1,292 @@
-# node-modules-patterns
\ No newline at end of file
+# Module Patterns in Node
+
+## Overview
+
+Imagine a scenario where you have an application with 2,000 lines of code. All the code is in one file. That's a nightmare to navigate around and also difficult for different developers to collaborate without merge conflicts! What if you split the logic into 20 files which each file having 100 lines of code roughly. Much better right? So what's the best way to do that? The answer is modules!
+
+Modules allow developers to build better systems because the code can be separated into different files which are grouped into packages or modules. You can write your own module or use a module from an open-source community.
```

Comment:
We're using the word "module" to mean at least two different things in this paragraph ("package" vs. "thing you download from an open-source community") -- can we make it a bit clearer that when we say "module" (in Node.js, to keep things simple), we mean a chunk of code that may or may not span multiple files and that provides specific functionality? We can also make an analogy to Ruby gems, if that's helpful.